client-spark/common/src/main/java/org/apache/spark/shuffle/writer/WriteBufferManager.java

```diff
@@ -413,10 +414,9 @@ protected ShuffleBlockInfo createShuffleBlock(int partitionId, WriterBuffer wb)
 
   // it's run in single thread, and is not thread safe
```

Comment:
Is this comment incorrect? Could you help describe more about this bug? @lwllvyb

Reply:
> Is this comment incorrect? Could you help describe more about this bug? @lwllvyb

Base on the test results, this comment is incorrect.

Reply:
Let me revisit this part code

Reply:
I'm OK for this code change because it looks reasonable. But I hope we could find the root cause